### PR TITLE
Use kelektiv/node-uuid approach to generate UUIDs v4

### PR DESCRIPTION
**Pull request by kosdiamantis**
Original code found on:
https://github.com/kelektiv/node-uuid/tree/v3.3.3

### CODE REVIEW

File: src/helpers/uuid_helper.coffee [hunk 1]
```diff
@@ -1,35 +1,89 @@
 ###
 Tiny library to generate version 4 UUIDs
```
Comment (alistat) [inline src/helpers/uuid_helper.coffee]: It's not that tiny anymore :rofl:

File: src/helpers/uuid_helper.coffee [hunk 2]
```diff
@@ -1,35 +1,89 @@
...
-        # unless placeholder is 8,
-        # in which case a random number from 8 to 11
-        # in hexadecimal
-        (placeholder ^ random() * 16 >> placeholder/4).toString(16)
-      else
-        # or otherwise a template string replacing
-        # zeroes, ones, and eights with random hex digits
-        '10000000-1000-4000-8000-100000000000'.replace(/[018]/g, UUID.generate)
+    # WHATWG crypto RNG - http://wiki.whatwg.org/wiki/Crypto
+    whatwgRNG: () ->
```
Comment (alistat) [inline src/helpers/uuid_helper.coffee]: Do we need to export these functions?
(assuming that bundling works properly and they are not exported anyway)

File: src/helpers/uuid_helper.coffee [hunk 3]
```diff
@@ -1,35 +1,86 @@
...
+        rnds[i] = r >>> (i & 0x03) << 3 & 0xff
+
+      rnds
+
+    # _bytesToUuid() and _byteToHex were taken from:
+    # https://github.com/kelektiv/node-uuid/blob/v3.3.3/lib/bytesToUuid.js
+    #
+    # Convert array of 16 byte values to UUID string format of the form:
+    # XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+    _byteToHex: (byteToHex = []; byteToHex[i] = (i + 0x100).toString(16).substr(1) for i in [0...256])
```
Comment (alistat) [inline src/helpers/uuid_helper.coffee]: `byteToHex` and `i` are leaked (`payload.js:3587`)

Personally, I think this is fine

this-is-fine 0

Comment (kosdiamantis) [reply src/helpers/uuid_helper.coffee]: Nice catch!
I used a self invoking function.

**Pull request merged**